core/src/main/java/org/apache/druid/data/input/impl/MapInputRowParser.java

```diff
@@ -75,7 +76,7 @@ public MapInputRowParser(
       throw new ParseException(e, "Unparseable timestamp found! Event: %s", theMap);
     }
 
-    return ImmutableList.of(new MapBasedInputRow(timestamp.getMillis(), dimensions, theMap));
+    return ImmutableList.of(new MapBasedInputRow(timestamp, dimensions, theMap));
```

Comment:
There is no need to have such additional call `timestamp.getMillis()` , so also modified this place.